lib/demo_web/controllers/user_registration_controller.ex

```diff
@@ -0,0 +1,30 @@
+defmodule DemoWeb.UserRegistrationController do
+  use DemoWeb, :controller
+
+  alias Demo.Accounts
+  alias Demo.Accounts.User
+  alias DemoWeb.UserAuth
+
+  def new(conn, _params) do
+    changeset = Accounts.change_user_registration(%User{})
+    render(conn, "new.html", changeset: changeset)
+  end
+
+  def create(conn, %{"user" => user_params}) do
+    case Accounts.register_user(user_params) do
+      {:ok, user} ->
+        {:ok, _} =
+          Accounts.deliver_user_confirmation_instructions(
+            user,
+            &Routes.user_confirmation_url(conn, :confirm, &1)
+          )
+
+        conn
+        |> put_flash(:info, "User created successfully.")
+        |> UserAuth.login_user(user)
```

Comment:
Am I missing something here? Should the user not be logged in until they click confirmation link in the email?

Reply:
It depends on the app. Many allow unconfirmed access but blocks some actions until confirmation. So you should further customize this based on your requirements.

Reply:
I just ended up modifying the session create action to achieve this. 
```
 def create(conn, %{"user" => user_params}) do
    %{"email" => email, "password" => password} = user_params

    if user = Accounts.get_user_by_email_and_password(email, password) do
      if user.confirmed_at == nil do
        redirect(conn, to: Routes.user_confirmation_path(conn, :new))
      else
        UserAuth.login_user(conn, user, user_params)
      end
    else
      render(conn, "new.html", error_message: "Invalid e-mail or password")
    end
  end
```
I guess I could avoid the multi nested conditions. 
Not sure if its the best approach thoughts?